import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java

```diff
@@ -56,54 +65,72 @@ protected void store(
       PDVInputStream inDicomStream,
       Attributes response)
       throws DicomServiceException, IOException {
-    String sopClassUID = request.getString(Tag.AffectedSOPClassUID);
-    String sopInstanceUID = request.getString(Tag.AffectedSOPInstanceUID);
-    String transferSyntax = presentationContext.getTransferSyntax();
-    String remoteAeTitle = association.getCallingAET();
-
-    InputStream inBuffer =
-        DicomStreamUtil.dicomStreamWithFileMetaHeader(
-            sopInstanceUID, sopClassUID, transferSyntax, inDicomStream);
-
-    MultipartContent content = new MultipartContent();
-    content.setMediaType(new HttpMediaType("multipart/related; type=\"application/dicom\""));
-    content.setBoundary(UUID.randomUUID().toString());
-    InputStreamContent dicomStream = new InputStreamContent("application/dicom", inBuffer);
-    content.addPart(new MultipartContent.Part(dicomStream));
-
-    HttpResponse httpResponse = null;
-    GenericUrl url = new GenericUrl(apiAddr + path);
     try {
-      HttpRequest httpRequest = requestFactory.buildPostRequest(url, content);
-      httpResponse = httpRequest.execute();
-    } catch (IOException e) {
-      e.printStackTrace();
-      throw new DicomServiceException(Status.ProcessingFailure, e);
-    }
+      MonitoringService.addEvent(Event.CSTORE_REQUEST);
+
+      String sopClassUID = request.getString(Tag.AffectedSOPClassUID);
+      String sopInstanceUID = request.getString(Tag.AffectedSOPInstanceUID);
+      String transferSyntax = presentationContext.getTransferSyntax();
+      String remoteAeTitle = association.getCallingAET();
+
+      validateParam(sopClassUID, "AffectedSOPClassUID");
+      validateParam(sopInstanceUID, "AffectedSOPInstanceUID");
+
+      CountingInputStream countingStream = new CountingInputStream(inDicomStream);
+      InputStream inBuffer =
+          DicomStreamUtil.dicomStreamWithFileMetaHeader(
+              sopInstanceUID, sopClassUID, transferSyntax, countingStream);
 
-    if (!httpResponse.isSuccessStatusCode()) {
-      String errorMessage = String.format(
-              "Got error response code for STOW-RS POST: %d, %s",
-              httpResponse.getStatusCode(), httpResponse.getStatusMessage());
```

Comment:
Why remove this logic? Don't we still need to test the status code?

Reply:
It's unreachable code, because httpRequest.execute() throws HttpResponseException if !isSuccessStatusCode() . Which is already processed in CStoreService.